controller/tap/server.go

```diff
@@ -357,6 +383,22 @@ func (s *GRPCTapServer) translateEvent(orig *proxy.TapEvent) *public.TapEvent {
 			}
 		}
 
+		headers := func(orig *httpPb.Headers) *public.Headers {
+			var headers []*public.Headers_Header
+			for _, header := range orig.GetHeaders() {
+				n := header.GetName()
```

Comment:
Just want to point out that technically, it is possible for `orig.GetHeaders()` to return a nil. Is it possible for a `TapEvent` to have no headers?

Reply:
@ihcsim Yep it is possible for a `TapEvent` to have no headers. For example it's common in `ResonseEndEvent`s

```
  "responseEndEvent": {
    "id": {
      "base": 0,
      "stream": 22
    },
    "sinceRequestInit": {
      "nanos": 8529614
    },
    "sinceResponseInit": {
      "nanos": 12929
    },
    "responseBytes": 0,
    "trailers": null,
    "grpcStatusCode": 0
  }
}
```

When `orig.GetHeaders()` returns `nil`, then it shows as above.